components/Buttons/src/MDCFloatingButton.m

```diff
@@ -134,12 +150,64 @@ - (void)encodeWithCoder:(NSCoder *)aCoder {
   [aCoder encodeInteger:self.mode forKey:MDCFloatingButtonModeKey];
   [aCoder encodeInteger:self.imageLocation forKey:MDCFloatingButtonImageLocationKey];
   [aCoder encodeDouble:self.imageTitleSpacing forKey:MDCFloatingButtonImageTitleSpacingKey];
+  [aCoder encodeObject:self.shapeToModeToMinimumSize
+                forKey:MDCFloatingButtonMinimumSizeDictionaryKey];
+  [aCoder encodeObject:self.shapeToModeToMaximumSize
+                forKey:MDCFloatingButtonMaximumSizeDictionaryKey];
+  [aCoder encodeObject:self.shapeToModeToContentEdgeInsets
+                forKey:MDCFloatingButtonContentEdgeInsetsDictionaryKey];
+  [aCoder encodeObject:self.shapeToModeToHitAreaInsets
+                forKey:MDCFloatingButtonHitAreaInsetsDictionaryKey];
 }
 
 - (void)commonMDCFloatingButtonInit {
-  _mode = MDCFloatingButtonModeNormal;
-  _imageLocation = MDCFloatingButtonImageLocationLeading;
   _imageTitleSpacing = MDCFloatingButtonDefaultImageTitleSpacing;
+
+  const CGSize miniSizeNormal = CGSizeMake(MDCFloatingButtonMiniDimension,
+                                           MDCFloatingButtonMiniDimension);
+  const CGSize defaultSizeNormal = CGSizeMake(MDCFloatingButtonDefaultDimension,
+                                              MDCFloatingButtonDefaultDimension);
+
+  NSMutableDictionary *miniMinimumSizes =
```

Comment:
Nit: Since it's hard for someone not very familiar with this code to wrap your head around what `miniMinimum` actually means, I'd add a comment explaining what this section is about ("Minimum button sizes for the mini shape in both normal and expanded modes") to improve readability.

Reply:
Done.